tools/m2kcli/commands/communication/i2c.cpp

```diff
@@ -0,0 +1,189 @@
+	Validator::validate(arguments["scl"], "scl", scl);
+
+	int sda;
+	Validator::validate(arguments["sda"], "sda", sda);
+
+	if (arguments.count("write_only")) {
+		Validator::validate(arguments["write_only"], "write_only", writeOnly);
+	}
+
+	m2k_i2c_init m2KI2CInit;
+	m2KI2CInit.scl = scl;
+	m2KI2CInit.sda = sda;
+	m2KI2CInit.context = context;
+
+	i2c_init_param i2CInitParam;
+	i2CInitParam.max_speed_hz = frequency;
+	i2CInitParam.slave_address = address;
+	i2CInitParam.extra = (void *) &m2KI2CInit;
+
+	uint32_t retVal;
+
+	if (writeOnly) {
+		retVal = i2c_init_write_only(&i2cDesc, &i2CInitParam);
+	} else {
+		retVal = i2c_init(&i2cDesc, &i2CInitParam);
+	}
+
+	if (retVal != 0) {
+		throw std::runtime_error("Could not initiate I2C\n");
+	}
+}
+
+void I2c::handleWrite()
+{
+	std::map<std::string, std::string> arguments = Validator::validate(getArguments());
+	if (!(arguments.count("data") && arguments.count("option"))) {
+		throw std::runtime_error("Expecting: data=<value>,... option=<value>\n");
+	}
+
+	std::vector<uint8_t> data;
+	Validator::validate(arguments["data"], "data", data);
+
+	int option;
+	Validator::validate(arguments["option"], "option", option);
+
+	if (writeOnly) {
+		i2c_write_only(i2cDesc, data.data(), data.size(), option);
+	} else {
+		i2c_write(i2cDesc, data.data(), data.size(), option);
+	}
+}
+
+void I2c::handleRead()
+{
+	std::map<std::string, std::string> arguments = Validator::validate(getArguments());
+	if (!(arguments.count("bytes_number") && arguments.count("option"))) {
+		throw std::runtime_error("Expecting: bytes_number=<value> option=<value>\n");
+	}
+
+	int bytesNumber;
+	Validator::validate(arguments["bytes_number"], "bytes_number", bytesNumber);
+
+	int option;
+	Validator::validate(arguments["option"], "option", option);
+
+	std::vector<uint8_t> data(bytesNumber);
+	i2c_read(i2cDesc, data.data(), bytesNumber, option);
+	for (auto &byte : data) {
+		std::cout << "0x" << std::hex << std::uppercase<< (int) byte << std::endl;
+	}
+}
+
+const struct option I2c::options[] = {
+	{"help",  no_argument,       nullptr, 'h'},
+	{"init",  required_argument, nullptr, 'i'},
+	{"write", required_argument, nullptr, 'w'},
+	{"read",  required_argument, nullptr, 'r'},
+	{nullptr, 0,                 nullptr, 0},
+};
+
+const char *const I2c::helpMessage = "Usage:\n"
+				     "m2kcli i2c <uri>\n"
+				     "           [-h | --help]\n"
+				     "           [-i | --init frequency=<value> address=<value> scl=<index> sda=<index>]\n"
+				     "           [-w | --write data=<value>,... option=<value> [write_only=<value>]]\n"
+				     "           [-r | --read bytes_number=<value> option=<value>]\n"
+				     "\n"
+				     "Positional arguments:\n"
+				     "  uri                   describe the context location \n"
+				     "Optional arguments:\n"
+				     "  -h, --help            show this help message and exit\n"
+				     "  -i, --init [frequency=<value> address=<value> scl=<index> sda=<index> [write_only=<value>]]\n"
```

Comment:
If we do something like: `m2kcli i2c ip:192.168.2.1 -i frequency=100000 address=0x48 scl=0 sda=1 write_only=1 -r bytes_number=1 option=1` will it crash to tell you something is wrong?

Reply:
Validation added in commit cb75049.